lib/url.c

```diff
@@ -4028,6 +4034,14 @@ static CURLcode create_conn(struct Curl_easy *data,
    *************************************************************/
   result = resolve_server(data, conn, async);
 
+  /* Strip trailing dots. */
+  strip_trailing_dot(&conn->host);
+  if(conn->bits.httpproxy)
+    strip_trailing_dot(&conn->http_proxy.host);
+  if(conn->bits.socksproxy)
+    strip_trailing_dot(&conn->socks_proxy.host);
+  // not dot-stripping: conn->conn_to_host->name
```

Comment:
``conn->conn_to_host->name`` may be used in CONNECT requests to HTTP proxies, so this also needs dot-stripping.

Reply:
Indeed. In Curl_proxy_connect a `Host:` is generated from it.
Pushed.